### ISSUE TITLE
Wayland from scratch - unhandled event when running under Debian 12 "bookworm"

### ISSUE THREAD

**Issue by jimsbugs** [gaultier/blog#10]
After addressing Issue #9, running `wayland.c` on Debian 12 on an x86_64 gives the following output:

```
$ ./wayland 
-> wl_display@1.get_registry: wl_registry=2
<- wl_registry@2.global: name=1 interface=wl_compositor version=5
-> wl_registry@2.bind: name=1 interface=wl_compositor version=5
<- wl_registry@2.global: name=2 interface=wl_shm version=1
-> wl_registry@2.bind: name=2 interface=wl_shm version=1
<- wl_registry@2.global: name=3 interface=wl_output version=3
<- wl_registry@2.global: name=4 interface=zxdg_output_manager_v1 version=3
<- wl_registry@2.global: name=5 interface=wl_data_device_manager version=3
<- wl_registry@2.global: name=6 interface=zwp_primary_selection_device_manager_v1 version=1
<- wl_registry@2.global: name=7 interface=wl_subcompositor version=1
<- wl_registry@2.global: name=8 interface=xdg_wm_base version=4
-> wl_registry@2.bind: name=8 interface=xdg_wm_base version=4
<- wl_registry@2.global: name=9 interface=gtk_shell1 version=5
<- wl_registry@2.global: name=10 interface=wp_viewporter version=1
<- wl_registry@2.global: name=11 interface=zwp_pointer_gestures_v1 version=3
<- wl_registry@2.global: name=12 interface=zwp_tablet_manager_v2 version=1
<- wl_registry@2.global: name=13 interface=wl_seat version=8
<- wl_registry@2.global: name=14 interface=zwp_relative_pointer_manager_v1 version=1
<- wl_registry@2.global: name=15 interface=zwp_pointer_constraints_v1 version=1
<- wl_registry@2.global: name=16 interface=zxdg_exporter_v1 version=1
<- wl_registry@2.global: name=17 interface=zxdg_importer_v1 version=1
<- wl_registry@2.global: name=18 interface=zwp_linux_dmabuf_v1 version=3
<- wl_registry@2.global: name=19 interface=wp_single_pixel_buffer_manager_v1 version=1
<- wl_registry@2.global: name=20 interface=zwp_keyboard_shortcuts_inhibit_manager_v1 version=1
<- wl_registry@2.global: name=21 interface=zwp_text_input_manager_v3 version=1
<- wl_registry@2.global: name=22 interface=wp_presentation version=1
<- wl_registry@2.global: name=23 interface=xdg_activation_v1 version=1
-> wl_compositor@3.create_surface: wl_surface=6
-> xdg_wm_base@5.get_xdg_surface: xdg_surface=7 wl_surface=6
-> xdg_surface@7.get_toplevel: xdg_toplevel=8
-> wl_surface@6.commit: 
<- wl_shm: format=0
<- wl_shm: format=0x1
<- wl_shm: format=0x34324241
<- wl_shm: format=0x34324258
<- wl_shm: format=0x36314752
<- wl_shm: format=0x30335241
<- wl_shm: format=0x30335258
<- wl_shm: format=0x30334241
<- wl_shm: format=0x30334258
<- wl_shm: format=0x48345241
<- wl_shm: format=0x48345258
<- wl_shm: format=0x48344241
<- wl_shm: format=0x48344258
object_id=8 opcode=2 msg_len=40
wayland: wayland.c:643: wayland_handle_message: Assertion `0 && "todo"' failed.
Aborted
```
If my understanding is correct, this is a `configure_bounds` event being sent to the `xdg_toplevel` object which has two integer arguments, a width and a height.  The following patch handles this event by consuming the two arguments and logging the event like the other cases.

```
$ diff -u wayland.c.orig  wayland.c 
--- wayland.c.orig	2023-10-15 19:21:27.000000000 -0700
+++ wayland.c	2023-10-17 11:50:36.430886532 -0700
@@ -29,6 +29,7 @@
 static const uint16_t wayland_xdg_wm_base_event_ping = 0;
 static const uint16_t wayland_xdg_toplevel_event_configure = 0;
 static const uint16_t wayland_xdg_toplevel_event_close = 1;
+static const uint16_t wayland_xdg_toplevel_event_configure_bounds = 2;
 static const uint16_t wayland_xdg_surface_event_configure = 0;
 static const uint16_t wayland_wl_display_get_registry_opcode = 1;
 static const uint16_t wayland_wl_registry_bind_opcode = 0;
@@ -633,6 +634,15 @@
 
     return;
   } else if (object_id == state->xdg_toplevel &&
+             opcode == wayland_xdg_toplevel_event_configure_bounds) {
+    uint32_t w = buf_read_u32(msg, msg_len);
+    uint32_t h = buf_read_u32(msg, msg_len);
+
+    printf("<- xdg_toplevel@%u.configure_bounds: w=%u h=%u\n",
+           state->xdg_toplevel, w, h);
+
+    return;
+  } else if (object_id == state->xdg_toplevel &&
              opcode == wayland_xdg_toplevel_event_close) {
     printf("<- xdg_toplevel@%u.close\n", state->xdg_toplevel);
     exit(0);
```
With this patch the program now runs correctly on my system - hopefully I've done the right thing here.

I should note that I knew nothing about the mechanics of writing a Wayland program prior to reading your tutorial - thanks very much for making it available!

**Comment by gaultier** [comment]
Hello, thanks for the comment.

As for the events: indeed, to keep it short and simple, I have only handled the absolute minimum for my compositor (sway). I should also mention that other compositors might send more events.
The nice thing is, some events can be read and ignored (but some need a reply such as ping/pong).

I have added a note in the article in that sense: db16bf54c0ebfef6a53340cc5aab325b4a544496
